pysmt/solvers/yices.py

```diff
@@ -261,27 +263,41 @@ def get_value(self, item):
         titem = self.converter.convert(item)
         ty = self.environment.stc.get_type(item)
         if ty.is_bool_type():
-            status, res = yicespy.yices_get_bool_value(self.model, titem)
-            self._check_error(status)
+            ytval = yices_api.c_int32()
+            errcode = yices_api.yices_get_bool_value(self.model, titem, ytval)
+            self._check_error(errcode)
+            res = ytval.value
             return self.mgr.Bool(bool(res))
         elif ty.is_int_type():
-            res = yicespy.yices_get_integer_value(self.model, titem)
-            return self.mgr.Int(res)
+            mpz = yices_api.mpz_t()
+            errcode = yices_api.yices_get_mpz_value(self.model, titem, mpz)
+            self._check_error(errcode)
+            mpz_term = yices_api.yices_mpz(mpz)
+            s = yices_api.yices_term_to_string(mpz_term, PRINTING_WIDTH, 1, 0)
+            return self.mgr.Int(int(s))
```

Comment:
(Not introduced here). These are a lot of steps going through the API! Do you have any idea of the performance of `yices_api`?

Reply:
It is quite fast, but I do not have benchmark numbers... Here we basically print an `mpz_t` to avoid finite-precision arithmetic errors (overflows in particular). I could be possible to directly access `mpz` without creating the `yices_mpz` term, but to do so we would need to import and directly use GMP API via ctypes...